geps/gep-3779/index.md

```diff
@@ -42,20 +37,432 @@ Kubernetes core provides NetworkPolicies as one way to do it. Network Policies h
+    // implementation. Common methods include:
+    // - From peer mTLS certificates: The identity is extracted from the client's
+    //   mTLS certificate presented during connection establishment.
+    // - From IP-to-identity mappings: The implementation might maintain a dynamic
+    //   mapping between source IP addresses (pod IPs) and their associated
+    //   identities (e.g., Service Account, SPIFFE IDs).
+    // - From JWTs or other request-level authentication tokens.
+    //
+    // Note for reviewers: While this GEP primarily focuses on identity-based
+    // authorization where identity is often established at the transport layer,
+    // some implementations might derive identity from authenticated tokens or sources
+    // within the request itself.
+    //
+    // +optional
+    Identities []string `json:"identities,omitempty"`
+
+    // ServiceAccounts specifies a list of Kubernetes Service Accounts that are
+    // matched by this rule. A request originating from a pod associated with
+    // one of these Serviceaccounts will match the rule.
+    //
+    // Values must be in one of the following formats:
+    //   - "<namespace>/<serviceaccount-name>": A specific Serviceaccount in a namespace.
+    //   - "<namespace>/*": All Serviceaccounts in the given namespace.
+    //   - "<serviceaccount-name>": a Serviceaccount in the same namespace as the policy.
+    //
+    // Use of "*" alone (i.e., all Serviceaccounts in all namespaces) is not allowed.
+    // To select all Serviceaccounts in the current namespace, use "<namespace>/*" explicitly.
+    //
+    // Example:
+    //   - "default/bookstore" → Matches Serviceaccount "bookstore" in namespace "default"
+    //   - "payments/*" → Matches any Serviceaccount in namespace "payments"
+    //   - "frontend" → Matches "frontend" Serviceaccount in the same namespace as the policy
+    //
+    // The ServiceAccounts listed here are expected to exist within the same
+    // trust domain as the targeted workload, which in many environments means
+    // the same Kubernetes cluster. Cross-cluster or cross-trust-domain access
+    // should instead be expressed using the `Identities` field.
+    // 
+    // +optional 
+    ServiceAccounts []string `json:"serviceAccounts,omitempty"`
+}
+
+// AuthorizationAttribute defines L4 properties of a request destination.
+type AuthorizationNetworkAttributes struct {
+    // Ports specifies a list of destination ports to match on.
+    // Traffic is matched if it is going to any of these ports.
+    // If not specified, the rule applies to all ports.
+    // +optional
+    Ports []gatewayv1.PortNumber `json:"ports,omitempty"`
+}
+
+```
+
+Note: Existing AuthorizationAPIs recognized the need to support negation fields like `not{field}`. To avoid duplicating fields with negation, we plan to support richer match expressions for fields in `AuthorizationSource` such as `matchExpressions: { operator: In|NotIn, values: []}`.
```

Comment:
Another "out of scope for now" bit. "More advanced logic like negation or explicit Boolean operations is left for potential future work" maybe.